sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java

```diff
@@ -241,7 +244,10 @@ public Formula getFormula() {
    *
    * @param formula the formula
    * @return this
+   * @deprecated since 5.2 there's no more decorator on batch side
+   * TODO add link to no API
    */
+  @Deprecated
   public Metric<G> setFormula(Formula formula) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)